tests/integration/mixer/policy/white_black_listing_test.go

```diff
@@ -23,7 +23,66 @@ import (
 	util "istio.io/istio/tests/integration/mixer"
 )
 
+func validateIPRule(t *testing.T, ctx framework.TestContext) {
+	t.Helper()
+	if galInst == nil {
+		t.Fatalf("galley not setup")
+	}
+	g := *galInst
+	if bookinfoNamespace == nil {
+		t.Fatalf("bookinfo namespace not allocated in setup")
+	}
+	bookinfoNs := *bookinfoNamespace
+	g.ApplyConfigOrFail(
+		t,
+		bookinfoNs,
+		bookinfo.NetworkingBookinfoGateway.LoadGatewayFileWithNamespaceOrFail(t, bookinfoNs.Name()))
+	g.ApplyConfigOrFail(
+		t,
+		bookinfoNs,
+		bookinfo.GetDestinationRuleConfigFile(t, ctx).LoadWithNamespaceOrFail(t, bookinfoNs.Name()),
+		bookinfo.NetworkingVirtualServiceAllV1.LoadWithNamespaceOrFail(t, bookinfoNs.Name()),
+	)
+	defer g.DeleteConfigOrFail(
+		t,
+		bookinfoNs,
+		bookinfo.NetworkingBookinfoGateway.LoadGatewayFileWithNamespaceOrFail(t, bookinfoNs.Name()),
+	)
+	defer g.DeleteConfigOrFail(
+		t,
+		bookinfoNs,
+		bookinfo.GetDestinationRuleConfigFile(t, ctx).LoadWithNamespaceOrFail(t, bookinfoNs.Name()),
+		bookinfo.NetworkingVirtualServiceAllV1.LoadWithNamespaceOrFail(t, bookinfoNs.Name()),
+	)
+
+	if ingInst == nil {
```

Comment:
is this dependent on config application?  If not, should we check it first, and fail fast?  Or is there some sort of timing issue at play?

Reply:
There is a TestMain that is shared by all tests here. This checks that the Main has run.